packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -96,6 +106,7 @@ export class DockerImageAsset extends Construct implements assets.IAsset {
       directoryName: staging.stagedPath,
       dockerBuildArgs: props.buildArgs,
       dockerBuildTarget: props.target,
+      dockerFile: file,
```

Comment:
This needs to be a relative path since the docker directory is copied into cloud assembly ("staging"). Since we resolve `dir` above, it results in an absolute path.

Reply:
I think `props.file` would just work here

Reply:
This was actually done intentionally. The idea was to do the validation as soon as possible (like the other validations). To validate - I have to `join` the directory and the file. If I were to pass only the relative path here, it means the cli would have to `join` again - resulting in duplicate logic. I think its also inline with the notion of letting the framework do the work and keeping the cli as simple as possible. 

I think what I missed though is that the final command would look something like:

```console
docker build --file ${props.directory}/${props.file} /generated/path/to/cloudassemby/....
```

Which means the command relies on resources outside the cloudassemly. Is this what you meant by:

> needs to be a relative path since the docker directory is copied into cloud assembly ?

Reply:
Ok that the validation is done early, but the value passed to "docker build" needs to be relative to the docker directory.

Reply:
Actually the value can't be relative because docker will fail - it doesn't assume you pass a relative path to the context when you use `--file`.

The path passed to `docker build --file` should be absolute, it just needs to be derived from the asset path inside the cloud assembly. What has to be relative is the property passed to the asset, since otherwise, the `manifest` in the cloud-assembly will contain an absolute path of the machine the assembly was created on.